cmd/bb_replicator/main.go

```diff
@@ -57,15 +58,18 @@ func main() {
 		log.Fatal("Failed to create replicator: ", err)
 	}
 
+	readyGroup := &sync.WaitGroup{}
+	readyGroup.Add(1)
 	go func() {
 		log.Fatal(
 			"gRPC server failure: ",
 			bb_grpc.NewServersFromConfigurationAndServe(
+				readyGroup,
```

Comment:
The alternative I see is to do
```golang
startupCtx, cancleStarupCtx := context.WithCancel()
grpcServerErrorChannel := bb_grpc.NewServersFromConfigurationAndServe(startupCtx, ...)
select {
case err := <-grpcServerErrorChannel:
    // Busy ports or other configuration errors will be reported here.
    log.Fatal("gRPC server failure: ", err)
default:
    break
}
go func() {
    log.Fatal("gRPC server failure: ", <-grpcServerErrorChannel)
}()
cancelStartup()
lifecycleState.MarkReadyAndWait()
```
This seems to be a bit bloated code compared to how it looks like today, but it is more clear what is happening. To make things more tidy, the error wrapping should be moved to a helper function.